docs/design/design.md

```diff
@@ -0,0 +1,131 @@
+# Dragonfly Design Doc
+
+## Contents
+
+- [API Design](#api-design)
+  - [Reference Docs](#reference-docs)
+- [Architecture](#architecture)
+- [Workflows](#workflows)
+  - [Dfget and Supernode's communication workflows](#dfget-and-supernode's-communication-workflows)
+  - [Supernode internal component workflows](#supernode-internal-component-workflows)
+
+## Nouns/Phase Explanation
+
+- **SuperNode**: SuperNode is a long-time process with two primary responsibilities:
+  - It's the tracker and scheduler in the P2P network that choose appropriate downloading net-path for each peer.
+  - It's also a CDN server that caches downloaded data from source to avoid downloading same files repeatedly.
+- **Dfget**: Dfget is the client of Dragonfly used for downloading files. It's similar to using wget. At the same time, it also plays the role of peer, which can transfer data between each other in p2p network.
+- **Dfdaemon**: Dfdaemon establishes a proxy between container engine and registry, as a local long-running process. It filters out layer fetching requests from all requests send by container engine when pulling images, then it uses Dfget to downloading these layers.
+- **P2P**: Peer-to-Peer, a distributed application architecture, more info refers to [wiki](https://en.wikipedia.org/wiki/Peer-to-peer).
+- **Task**: A Task will store some meta info about the taskFile, pieces and something else.A Task has a one-to-one correspondence with a file on the disk which is identified by taskID. If you want to download a file from Supernode, you should register a task which means that tell the server about the file info you want to download before actually do it.
```

Comment:
Given that you also defined `DfgetTask`, we probably can call this one as XXTask to make it more clear? Otherwise It is not easy to differentiate these two items, likely, DfgetTasks can be abbreviated as task in some context?

One thing I can comment on is: We should make the document clear, and avoid using terms like `something else`(we should precisely say which information included in it).